src/cls/fifo/cls_fifo_types.h

```diff
@@ -382,38 +419,22 @@ struct info {
     }
 
     for (const auto& entry : update.journal_entries_add()) {
-      auto iter = journal.find(entry.part_num);
-      if (iter != journal.end() &&
-	  iter->second.op == entry.op) {
-	/* don't allow multiple concurrent (same) operations on the same part,
-	   racing clients should use objv to avoid races anyway */
-	return fmt::format("multiple concurrent operations on same part are not "
-			   "allowed, part num={}", entry.part_num);
-      }
-
-      if (entry.op == journal_entry::Op::create) {
-	tags[entry.part_num] = entry.part_tag;
+      if (sr::find(journal, entry) != journal.end()) {
+	continue;
```

Comment:
because of continue here and no early return with ECANCEL, when multi rgw instance try to race and create same part and set head, both will go and call write_header and increment the version value individually.
this will cause version mismatch in subsequent read_header where there is check for version

Reply:
Is the idea that we get into a state where we can't call `read_info()` and succeed because we're passing a version to it?

I think that could be resolved just by making sure we don't (since in all cases I can think of we want to replace the info we have.)

Reply:
> Is the idea that we get into a state where we can't call `read_info()` and succeed because we're passing a version to it?
> 
> I think that could be resolved just by making sure we don't (since in all cases I can think of we want to replace the info we have.)

but on contrary why should version be updated for same operations by 2 different rgw instances.
Yes the in read_header() version is an optional_param and if we skip that we can bypass the error, but still IMO we should not increment the version for same operations.
I updated the [tracker](https://tracker.ceph.com/issues/57562#note-73) with the error we are currently getting